contracts/exchangeIssuance/DEXAdapter.sol

```diff
@@ -254,31 +284,261 @@ abstract contract DEXAdapter {
+    {
+        require(_path.length == 2, "ExchangeIssuance: CURVE_WRONG_PATH_LENGTH");
+        (int128 i, int128 j) = _getCoinIndices(_pool, _path[0], _path[1], ICurveAddressProvider(_addresses.curveAddressProvider));
+
+        uint256 amountIn = _getAmountInCurve(
+            _pool,
+            i,
+            j,
+            _amountOut,
+            _addresses
+        );
+        require(amountIn <= _maxAmountIn, "ExchangeIssuance: CURVE_OVERSPENT");
+
+        if(_path[0] == ETH_ADDRESS){
+            IWETH(_addresses.weth).withdraw(amountIn);
+        }
+
+        uint256 returnedAmountOut = _exchangeCurve(i, j, _pool, amountIn, _amountOut, _path[0]);
+        require(_amountOut <= returnedAmountOut, "ExchangeIssuance: CURVE_UNDERBOUGHT");
+
+        if(_path[_path.length-1] == ETH_ADDRESS){
+            IWETH(_addresses.weth).deposit{ value: returnedAmountOut }();
+        }
+
+
+        return amountIn;
+    }
+    
+    function _exchangeCurve(
+        int128 _i,
+        int128 _j,
+        address _pool,
+        uint256 _amountIn,
+        uint256 _minAmountOut,
+        address _from
+    )
+        private
+        returns (uint256 amountOut)
+    {
+        ICurvePool pool = ICurvePool(_pool);
+        if(_from == ETH_ADDRESS){
+            amountOut = pool.exchange{value: _amountIn}(
+                _i,
+                _j,
+                _amountIn,
+                _minAmountOut
+            );
+        }
+        else {
+            IERC20(_from).approve(_pool, _amountIn);
+            amountOut = pool.exchange(
+                _i,
+                _j,
+                _amountIn,
+                _minAmountOut
+            );
+        }
+    }
+
+
+
+    /**
+     *  Calculate required input amount to get a given output amount via Curve swap
+     *
+     * @param _i            Index of input token as per the ordering of the pools tokens
+     * @param _j            Index of output token as per the ordering of the pools tokens
+     * @param _pool         Address of curve pool to use
+     * @param _amountOut    The amount of output token to be received
+     * @param _addresses    Struct containing relevant smart contract addresses.
+     *
+     * @return amountOut    The amount of output token obtained
+     */
+    function _getAmountInCurve(
+        address _pool,
+        int128 _i,
+        int128 _j,
+        uint256 _amountOut,
+        Addresses memory _addresses
+    )
+        public
+        view
+        returns (uint256)
+    {
+        CurvePoolData memory poolData = _getCurvePoolData(_pool, ICurveAddressProvider(_addresses.curveAddressProvider));
+
+        return ICurveCalculator(_addresses.curveCalculator).get_dx(
+            poolData.nCoins,
+            poolData.balances,
+            poolData.A,
+            poolData.fee,
+            poolData.rates,
+            poolData.decimals,
+            // TODO: Review correct value for "underlying" parameter
+            false,
```

Comment:
According to the parameter [documentation in the deployed contract ](https://etherscan.io/address/0xc1DB00a8E5Ef7bfa476395cdbcc98235477cDE4E#code#L158) this parameter indicates `Whether the coin is in raw or lent-out form`. 

For the stEth/Eth usecase it works this way and I don't see any use case where we would wan to swap a "lent-out" token via curve.

Therefore I'd like to keep this hardcoded. Alternatively we could add this as an additional field in the `swapData` struct to be controlled by the user, but I don't think that's necessary.

@ncitron : Let me know if you disagree

Reply:
Yeah I also agree on this